api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -479,6 +506,100 @@ public void getProgramByUuid_shouldFindObjectGivenValidUuid() {
 	public void getProgramByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
 		Assert.assertNull(Context.getProgramWorkflowService().getProgramByUuid("some invalid uuid"));
 	}
+
+	/**
+	 * @see ProgramWorkflowService#getPrograms(String)
+	 */
+	@Test
+	public void getPrograms_shouldGetProgramByNameFragment() {
+		List<Program> allPrograms = Context.getProgramWorkflowService().getAllPrograms();
+		int programCount=0;
+		final String programName = "program name";
+		for(Program program:allPrograms) {
+			if(program.getName().equals(programName)) {
+				programCount +=1;
+			}
+		}
+		List<Program> programs= Context.getProgramWorkflowService().getPrograms(programName);
+		assertEquals(programCount, programs.size());
+	}
+
+	/**
+	 * @see ProgramWorkflowService#unretireProgram(Program)
+	 */
+	@Test
+	public void unretireProgram_shouldUnRetireProgram() {
+		Program program = Context.getProgramWorkflowService().getProgram(1);
+		assertEquals(program.getRetired(), false);
+
+		Program retiredProgram = Context.getProgramWorkflowService().retireProgram(program,"for testing");
+		assertEquals(program.getRetired(), true);
+
+		Program unretireProgram = Context.getProgramWorkflowService().unretireProgram(retiredProgram);
+		assertEquals(unretireProgram.getRetired(), false);
+	}
+
+	/**
+	 * @see ProgramWorkflowService#saveConceptStateConversion(ConceptStateConversion)
+	 */
+	@Test(expected = APIException.class)
+	public void saveConceptStateConversion_shouldFaileIfRequiredParametersMissing() {
+		ConceptStateConversion conceptStateConversion = new ConceptStateConversion(1);
+		// all required parameters are missing
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
+
+		conceptStateConversion.setConcept(cs.getConcept(3));
+		// ProgramWorkflow, ProgramWorkflowState are missing
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
+
+		conceptStateConversion.setProgramWorkflow(pws.getWorkflow(1));
+		// ProgramWorkflowState is missing
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
+	}
+
+	/**
+	 * @see ProgramWorkflowService#saveConceptStateConversion(ConceptStateConversion)
+	 */
+	@Test
+	public void saveConceptStateConversion_shouldSaveIfAllRequiredParametersAreSatisfied() {
+		ConceptStateConversion conceptStateConversion = new ConceptStateConversion(1);
+		conceptStateConversion.setConcept(cs.getConcept(3));
+		conceptStateConversion.setProgramWorkflow(pws.getWorkflow(1));
+		conceptStateConversion.setProgramWorkflowState(pws.getWorkflow(1).getState(1));
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
```

Comment:
use the test class field `pws` instead of taking the service again from the `Context`.

insert newlines between arranging the `ConceptStateConversion` and `saveConceptStateConversion`.

and add expectations. what would you expect to happen?